azure-ai-search/index_search/__init__.py

```diff
@@ -47,21 +57,35 @@ class AzureIndexSearchConfig:
     index_name: str | None = None
     highlight_fields: str | None = None
     highlight_tag: str = "em"
+    result_transform_map: dict = field(
+        default_factory=lambda: {
+            "id": "/id",
+            "title": "/title",
+            "score": "/@search.score",
+            "subtitle": "/subtitle",
+            "url": "/url",
+            "content": "/@search.highlights/content/0",
+            "last_updated": "/last_updated",
+        }
+    )
 
 
-def transform_result(result):
-    content_highlights = (result["@search.highlights"] or {}).get(
-        "content", ["No content available"]
-    )
-    return {
-        "id": result.get("id"),
-        "url": result.get("url"),
-        "score": result.get("@search.score"),
-        "title": result.get("metadata_storage_name"),
-        "content": content_highlights[0],
-        "subtitle": result.get("subtitle"),
-        "last_updated": result.get("metadata_last_modified"),
-    }
+def transform(source_dict, path_map):
+    if path_map is None:
+        return {}
+
+    transformed_dict = {}
+    for new_key, path in path_map.items():
+        try:
+            value = dpath.get(source_dict, path)
+        except (KeyError, ValueError):
+            value = None
```

Comment:
are we ok to silently fail when an expected path is not found in the source dictionary? would have thought this would be an exception or at least an error log?

Reply:
No we shouldn't silently fail. Now raises a custom exception.